tests/Application/Persistence/RetrySafeEntityManagerTest.php

```diff
@@ -152,6 +152,65 @@ public function testFlushWithRetry(): void
         $this->retrySafeEntityManager->flush();
     }
 
+    public function testRefresh(): void
+    {
+        $underlyingEmProphecy = $this->prophesize(EntityManager::class);
+        $underlyingEmProphecy
+            ->refresh(Argument::type(Donation::class),LockMode::PESSIMISTIC_WRITE)
+            ->shouldBeCalledOnce();
+
+        $retrySafeEntityManagerReflected = new ReflectionClass($this->retrySafeEntityManager);
```

Comment:
I know this is following the pattern of the existing `testPersist`, but needing reflection in tests is often a sign that the subject under test could be better designed. Any reason why RetrySafeEntityManager shouldn't take its underlying entityManger via constructor injection so that it would be straight forward to replace in tests?

Reply:
I don't recall really as it was 2019, but think my thinking was probably that we want to ensure the first build, on construct, is guaranteed to happen exactly the same way as much rarer resets when they call `buildEntityManager()`. If the EM was created first that might not be true.